python/paddle/fluid/clip.py

```diff
@@ -264,15 +264,15 @@ def _process_context(self, context, param, grad):
             context[self.group_name] = []
             context[self.group_name + "_clip_value"] = self.clip_norm
             context[self.group_name + "_clip"] = layers.fill_constant(
-                shape=[1], dtype="float32", value=self.clip_norm)
+                shape=[1], dtype=grad.dtype, value=self.clip_norm)
```

Comment:
I think `dtype="float32"` is a hard code here, the type should be the same with grad, right? @ktlichkid